src/app/store/reducers/error.reducers.ts

```diff
@@ -0,0 +1,33 @@
+import { ErrorActions, EErrorActions } from '../actions/Error.actions';
+
+export interface ErrorState {
+  errors: string[] | null;
+}
+
+export const initialErrorState: ErrorState = {
+  errors: []
+};
+
+export function ErrorReducer(
+  state: ErrorState = initialErrorState,
+  action: ErrorActions
+): ErrorState {
+  switch (action.type) {
+    case EErrorActions.SetErrorMessage: {
+      return {
+        ...state,
+        errors: [...action.payload]
+      };
+    }
+
+    case EErrorActions.ClearErrorMessage: {
```

Comment:
shouldn't be null instead? i think we define the default value as null which is probably when we have no error?
Also, shouldn't initialErrorState be null instead of [] in fact?
I suspect that with the empty array we could still map over it hence the errorComponent could show with empty messages?

Reply:
This approach is preferred greatly because it gives better API flexibility in terms of checking for the existence of errors in the state. It also matches the expected response received from back end which are string array of errors

Reply:
Thanks for the clarification.
Shouldn't we have this instead (of null for the default state) if it is the case?
```
export interface ErrorState {
  errors: string[] | [];
}
```